lib/internal/process/esm_loader.js

```diff
@@ -58,11 +59,35 @@ async function initializeLoader() {
   const { getOptionValue } = require('internal/options');
   const customLoaders = getOptionValue('--experimental-loader');
   const preloadModules = getOptionValue('--import');
-  const loaders = await loadModulesInIsolation(customLoaders);
+
+  let cwd;
+  try {
+    cwd = process.cwd() + '/';
+  } catch {
+    cwd = 'file:///';
+  }
+
+  const internalEsmLoader = new ESMLoader();
+  const loaders = [];
+
+  for (let i = 0; i < customLoaders.length; i++) {
+    const customLoader = customLoaders[i];
+
+    // Importation must be handled by internal loader to avoid poluting userland
+    const keyedExportsSublist = await internalEsmLoader.import(
+      [customLoader],
+      pathToFileURL(cwd).href,
+      ObjectCreate(null),
+    );
```

Comment:
This diff makes it so that we load loaders one at a time (because each one must be used to resolve/loader the subsequent ones).

Why the wrapper: without it, `import` makes an [early return](https://github.com/nodejs/node/blob/40ab56cae1630af03498de48bdbe0259f1fc50ad/lib/internal/modules/esm/loader.js#L539) which shortcuts the pairing.